[App Config] Add tags filter & list by label methods for 2023-11-01 API version (#30161)
Azure/azure-sdk-for-js: sdk/appconfiguration/app-configuration/samples/v1/javascript/README.md

| [getSettingOnlyIfChanged.js][getsettingonlyifchanged]           | Demonstrates getting a setting only if it has changed from what you already have. (This allows your app to avoid downloading the contents of a setting if the value is unchanged.)                                                                                                                                                                                                                                |
| [listConfigurationSettings.js][listconfigurationsettings]       | Demonstrates listing multiple configuration settings using a filter for a key or label.                                                                                                                                                                                                                                                                                                                           |

> | [listLabels.js][listlabels]                                     | Demonstrates listing revisions for a configuration setting.                                                                                                                                                                                                                                                                                                                                                       |

Reviewer: ```suggestion
| [listLabels.js][listlabels]                                     | Demonstrates listing labels in the configuration setting store?                                                                                                                                                                                                                                                                                                                                                       |
```
